src/Corporation/ui/BuybackSharesModal.tsx

```diff
@@ -36,22 +38,14 @@ export function BuybackSharesModal(props: IProps): React.ReactElement {
 
   function buy(): void {
     if (disabled) return;
-    if (shares === null) return;
-    corp.numShares += shares;
-    if (isNaN(corp.issuedShares)) {
-      console.warn("Corporation issuedShares is NaN: " + corp.issuedShares);
-      console.warn("Converting to number now");
-      const res = corp.issuedShares;
-      if (isNaN(res)) {
-        corp.issuedShares = 0;
-      } else {
-        corp.issuedShares = res;
-      }
+    try {
+      BuyBackShares(corp, player, shares)
+      props.onClose();
```

Comment:
These should probably be outside of the try catch.

Reply:
hmm possible ill see what happens in the "onclose" but i think your right 👍